src/importexport/musicxml/internal/musicxml/exportxml.cpp

```diff
@@ -5123,14 +5122,21 @@ void ExportMusicXml::textLine(TextLineBase const* const tl, staff_idx_t staff, c
         type = "stop";
     }
 
-    if (hook) {
-        if (hookHeight < 0.0) {
-            lineEnd = "up";
-            hookHeight *= -1.0;
-        } else {
-            lineEnd = "down";
-        }
-        rest += QString(" end-length=\"%1\"").arg(hookHeight * 10);
+    QString lineEnd;
+    switch (hookType) {
+    case HookType::HOOK_90T:
+        lineEnd = "both";
```

Comment:
This looks much better, but now the `end-length` is not written anymore for T hook. Is that expected?

Reply:
That was intentional. But I can add it if you'd prefer.